testEnv.go

```diff
@@ -0,0 +1,62 @@
+package kommons
+
+import (
+	"github.com/flanksource/commons/deps"
+	"io/ioutil"
+	"k8s.io/client-go/rest"
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"strings"
+)
+
+var APIServerDefaultArgs = []string{
+	"--advertise-address=127.0.0.1",
+	"--etcd-servers={{ if .EtcdURL }}{{ .EtcdURL.String }}{{ end }}",
+	"--cert-dir={{ .CertDir }}",
+	"--insecure-port={{ if .URL }}{{ .URL.Port }}{{ end }}",
+	"--insecure-bind-address={{ if .URL }}{{ .URL.Hostname }}{{ end }}",
+	"--secure-port={{ if .SecurePort }}{{ .SecurePort }}{{ end }}",
+	// we're keeping this disabled because if enabled, default SA is missing which would force all tests to create one
+	// in normal apiserver operation this SA is created by controller, but that is not run in integration environment
+	//"--disable-admission-plugins=ServiceAccount",
+	"--service-cluster-ip-range=10.0.0.0/24",
+	"--allow-privileged=true",
+}
+
+func InstallTestBin(version string) (string, error) {
+	dir, err := ioutil.TempDir("", "kubebuilder-*")
```

Comment:
Code note would be great. 

Re tempdir, this might be called by someone running local tests repeatedly, so want to keep that overhead low. Might make sense to store it in project bin even.